### PR TITLE
fix: logging error

### PR DESCRIPTION
**Pull request by nnnkkk7**
## Related Issue
  - Info about Issue or bug

Closes: #1588

#### Describe the changes you've made
A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

I fixed how to log error.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by shivamsouravjha** [comment]
Hey @nnnkkk7 ! thanks for opening the PR but seems this isn't linked to an issue. 

You can create an issue and link this PR to the issue.

File: pkg/hooks/launch.go [hunk 1]
```diff
@@ -657,7 +658,7 @@ func getInodeNumber(pid int) uint64 {
 
 	f, err := os.Stat(filepath)
 	if err != nil {
-		fmt.Errorf("%v failed to get the inode number or namespace Id:", Emoji, err)
+		log.Errorf("%v failed to get the inode number or namespace Id: %w", Emoji, err)
 		return 0
 	}
 	// Dev := (f.Sys().(*syscall.Stat_t)).Dev
```
Comment (shivamsouravjha) [inline pkg/hooks/launch.go:664]: instead of using log function, pass the logger from ``launch.go```

Comment (nnnkkk7) [reply pkg/hooks/launch.go]: thanks! I fixed it.
[use logger](https://github.com/keploy/keploy/pull/1351/commits/96041c591a798a9c0e0485edf693ef6981996ef7)

**Comment by nnnkkk7** [comment]
@shivamsouravjha 
thank you!
I created the issue and apply your review!

File: pkg/hooks/launch.go [hunk 2]
```diff
@@ -620,12 +621,12 @@ func parseDockerCommand(dockerCmd string) (string, string, error) {
 	return containerName, networkName, nil
 }
 
-func getInodeNumber(pid int) uint64 {
-
+func (h *Hook) getInodeNumber(pid int) uint64 {
 	filepath := filepath.Join("/proc", strconv.Itoa(pid), "ns", "pid")
```
Comment (shivamsouravjha) [inline pkg/hooks/launch.go:625]: any reason to move `getInodeNumber` into hooks?

Comment (nnnkkk7) [reply pkg/hooks/launch.go:625]: Because I want to use `h.logger.Error` in this function.

Comment (shivamsouravjha) [reply pkg/hooks/launch.go:625]: You could rather pass it as a parameter

Comment (nnnkkk7) [reply pkg/hooks/launch.go]: ok!
I fixed it! [pass logger as a parameter](https://github.com/keploy/keploy/pull/1351/commits/7e75d9266f95ddc90c59a7b972e436eaa43dc218)

File: pkg/hooks/launch.go [hunk 3]
```diff
@@ -15,6 +14,8 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/cloudflare/cfssl/log"
+
 	"github.com/cilium/ebpf"
```
Comment (shivamsouravjha) [inline pkg/hooks/launch.go:19]: where is this being used?

**Review by shivamsouravjha — CHANGES_REQUESTED** [review]

File: pkg/hooks/launch.go (hunk 3 above)
Comment (nnnkkk7) [reply pkg/hooks/launch.go]: @shivamsouravjha 
Thank you!
I fixed it!

**Comment by charankamarapu** [comment]
This issue has been addressed through refactoring changes, so closing this PR.

**Pull request closed without merging**